### PR TITLE
Implement service

### LINKED ISSUE #26: NewService: Create GET handler for all existing news
- define a request url to retrieve all existing news
- create a new GET handler for that request
- create a database finder to retrieve all news using "plain" Hibernate (not Spring Data)
- separate persistence model and service API model

### PR DESCRIPTION
**Pull request by asiday**
Changes:
-Create REST service that uses Spring Data.
-Add get request.

Close #26

### CODE REVIEW

File: crawler/src/main/resources/application.properties [hunk 1]
```diff
@@ -6,4 +6,5 @@ spring.datasource.url=
 spring.datasource.username=
 spring.datasource.password=
 spring.datasource.driver-class-name=org.mariadb.jdbc.Driver
-spring.jpa.hibernate.ddl-auto=none
\ No newline at end of file
+spring.jpa.hibernate.ddl-auto=none
+server.port=
```
Comment (yambay) [inline crawler/src/main/resources/application.properties:10]: You can remove this property.

File: crawler/src/main/resources/schema.sql [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
-    author VARCHAR(100),
-    title VARCHAR(250),
-    description VARCHAR(500),
-    url VARCHAR(250),
-    url_to_image VARCHAR(250),
-    published_at VARCHAR(50) NOT NULL,
-    content VARCHAR(1000),
-    PRIMARY KEY (id)
+CREATE TABLE articles (
+                          id INT NOT NULL AUTO_INCREMENT,
```
Comment (yambay) [inline crawler/src/main/resources/schema.sql:3]: Please, make the indentation smaller to make code more readable

File: service/pom.xml [hunk 3]
```diff
@@ -40,8 +35,29 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
-	</dependencies>
+		<dependency>
```
Comment (yambay) [inline service/pom.xml:38]: Why did you move the "spring-boot-starter-web" dependency from above? Please organize dependencies into groups of productive and test dependencies

File: service/src/test/java/dev/asida/service/ServiceApplicationTests.java [hunk 4]
```diff
@@ -1,13 +1,31 @@
...
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
 
 @SpringBootTest
-class ServiceApplicationTests {
+@Transactional
+public class ServiceApplicationTests {
+
```
Comment (yambay) [inline service/src/test/java/dev/asida/service/ServiceApplicationTests.java:18]: Remove empty rows

**Review by yambay — CHANGES_REQUESTED** [review]

File: crawler/src/main/resources/application.properties [hunk 5]
```diff
@@ -6,4 +6,5 @@ spring.datasource.url=
 spring.datasource.username=
 spring.datasource.password=
 spring.datasource.driver-class-name=org.mariadb.jdbc.Driver
```
Comment (yambay) [inline crawler/src/main/resources/application.properties:8]: blub

File: crawler/src/main/resources/application.properties (hunk 1 above)
Comment (asiday) [reply crawler/src/main/resources/application.properties:10]: done

File: crawler/src/main/resources/schema.sql (hunk 2 above)
Comment (asiday) [reply crawler/src/main/resources/schema.sql:3]: done

File: service/pom.xml (hunk 3 above)
Comment (asiday) [reply service/pom.xml:38]: done

File: service/src/main/java/dev/asida/service/controllers/ArticleController.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+@RestController
+public class ArticleController {
+
+    private final ArticleRepository articleRepository;
+
+    public ArticleController(ArticleRepository articleRepository) {
+        this.articleRepository = articleRepository;
+    }
+
+    @GetMapping("/articles") // see result on localhost:XXXX/articles
```
[the remark this answers is not in the document]
Comment (asiday) [reply service/src/main/java/dev/asida/service/controllers/ArticleController.java:28]: deleted

File: service/src/main/java/dev/asida/service/data/Article.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
+package dev.asida.service.data;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+/**
+ * Create a domain object together with its attributes.
+ *
+ * @Entity is a JPA annotation to make this object ready for storage in a JPA-based data store.
```
[the remark this answers is not in the document]
Comment (asiday) [reply service/src/main/java/dev/asida/service/data/Article.java:10]: deleted

File: service/src/main/resources/schema.sql [hunk 8]
```diff
@@ -0,0 +1,12 @@
+DROP TABLE IF EXISTS articles;
+CREATE TABLE articles (
+                          id INT NOT NULL AUTO_INCREMENT,
```
[the remark this answers is not in the document]
Comment (asiday) [reply service/src/main/resources/schema.sql:3]: fixed

File: service/src/test/java/dev/asida/service/ServiceApplicationTests.java [hunk 9]
```diff
@@ -1,13 +1,31 @@
...
+
+    @Autowired
+    private ArticleRepository repository;
+
+    @Test
+    public void whenApplicationStarts_thenHibernateCreatesInitialRecords() {
+        List<Article> articles = repository.findAll();
+
```
[the remark this answers is not in the document]
Comment (asiday) [reply service/src/test/java/dev/asida/service/ServiceApplicationTests.java:26]: Removed

File: service/src/test/java/dev/asida/service/ServiceApplicationTests.java (hunk 4 above)
Comment (asiday) [reply service/src/test/java/dev/asida/service/ServiceApplicationTests.java:18]: removed
